### PR TITLE
Prepare for 2.3.2

### PR DESCRIPTION
**Pull request by xiki-tempula**
remove python 3.9 support

### CODE REVIEW

**Comment by xiki-tempula** [comment]
Looks like we cannot have any badge in the readme for security reasons. https://github.com/pypa/readme_renderer/issues/304#issuecomment-2259039895

File: setup.py [hunk 1]
```diff
@@ -41,9 +40,14 @@
     packages=find_packages("src"),
     package_dir={"": "src"},
     license="BSD",
-    long_description=open("README.md").read(),
+    long_description=open("README.md")
+    .read()
+    .replace(
```
Comment (orbeckst) [inline setup.py:45]: I thought this issue is only due to scale-factor. For other projects, badges seem to work ok when the scaling was removed and we don't have scaling here. When I locally test (after `python -m build`)
```
twine check dist/alchemlyb-2.3.1+5.g911ce53*
Checking dist/alchemlyb-2.3.1+5.g911ce53-py3-none-any.whl: PASSED
Checking dist/alchemlyb-2.3.1+5.g911ce53.tar.gz: PASSED
```
it looks fine without removing the badges.

What was your specific reason for removing the badges?

**Review by orbeckst — CHANGES_REQUESTED** [review]
Not sure if badges need removing.

File: setup.py (hunk 1 above)
Comment (orbeckst) [reply setup.py:45]: Did you get an error from pypi?

The other linked issues seem to say that twine check passes but then pypi rejects, although I don't quite understand why as they are saying that they don't allow inclusion of local files whereas our badges are all links.

**Comment by xiki-tempula** [comment]
here is the failed pipeline https://github.com/alchemistry/alchemlyb/actions/runs/10537263963/job/29198456637

**Comment by orbeckst** [comment]
Can you try replacing `.svg` in the README badges with `.png` and not removing them from the PyPi README? This may work without any intermediate conversion magic under the hood and keep our scripts clearer.

**Comment by xiki-tempula** [comment]
@orbeckst I don't think any of the badge support png format https://github.com/alchemistry/alchemlyb/tree/389-release-232-for-joss-publication

**Comment by orbeckst** [comment]
Ah well, thanks for trying it out — I had just manually replaced a svg -> png for I think zenodo  and that had worked.

SVG
[DOI](https://doi.org/10.5281/zenodo.12692737)
```
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.12692737.svg)](https://doi.org/10.5281/zenodo.12692737)
```

PNG
[DOI](https://doi.org/10.5281/zenodo.12692737)
```
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.12692737.png)](https://doi.org/10.5281/zenodo.12692737)
```

But for codecov it does not work, as you say:
https://codecov.io/gh/alchemistry/alchemlyb/branch/master/graph/badge.png gives indeed
```

Badge Handler

GET /gh/alchemistry/alchemlyb/branch/master/graph/badge.png

HTTP 404 Not Found
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "detail": "File extension should be one of [ svg || txt ]"
}
```

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+[![Zenodo DOI](https://zenodo.org/badge/68669096.png)](https://zenodo.org/badge/latestdoi/68669096) [![Documentation](https://readthedocs.org/projects/alchemlyb/badge/?version=latest)](http://alchemlyb.readthedocs.io/en/latest/) [![Build Status](https://github.com/alchemistry/alchemlyb/actions/workflows/ci.yaml/badge.png?branch=master)](https://github.com/alchemistry/alchemlyb/actions/workflows/ci.yaml) [![Code coverage](https://codecov.io/gh/alchemistry/alchemlyb/branch/master/graph/badge.png)](https://codecov.io/gh/alchemistry/alchemlyb) [![anaconda package](https://anaconda.org/conda-forge/alchemlyb/badges/version.png)](https://anaconda.org/conda-forge/alchemlyb)
```
Comment (orbeckst) [inline README.md:3]: thanks for trying it out — didn't work

File: setup.py [hunk 3]
```diff
@@ -41,9 +40,14 @@
     packages=find_packages("src"),
     package_dir={"": "src"},
     license="BSD",
-    long_description=open("README.md").read(),
+    long_description=open("README.md")
```
Comment (orbeckst) [inline setup.py:43]: please add a comment referencing this issue and the pypi issue so that we could remove the hack again once pypi fixes there stuff

**Review by orbeckst — CHANGES_REQUESTED** [review]
Let's go with your original solution. Just add a comment, please.

**Review by orbeckst — APPROVED** [review]
LGTM, let's remove the badges.

I'll let you squash-merge as part of the release.

**Comment by xiki-tempula** [comment]
@orbeckst Do I need to do something for zenodo or it is done automatically?

**Pull request merged**